include/engine/base64.hpp

```diff
@@ -16,12 +17,26 @@
 
 // RFC 4648 "The Base16, Base32, and Base64 Data Encodings"
 // See: https://tools.ietf.org/html/rfc4648
-// Implementation adapted from: http://stackoverflow.com/a/28471421
```

Comment:
The tips here on how to use the Boost serialization iterators are wrong and dangerous! Padding needs to happen _before_ iterating over stream!